backends/bmv2/simple_switch/simpleSwitch.cpp

```diff
@@ -810,7 +810,25 @@ Util::IJson* ExternConverter_log_msg::convertExternFunction(
     auto str = ctxt->conv->convert(mc->arguments->at(0)->expression);
     params->append(str);
     if (mc->arguments->size() == 2) {
-        auto val = ctxt->conv->convert(mc->arguments->at(1)->expression);
+        auto arg1 = mc->arguments->at(1)->expression;
+        // this must be a list expression, with all components
+        // evaluating to integral types.
+        auto argType = ctxt->typeMap->getType(arg1);
+        if (auto ts = argType->to<IR::Type_List>()) {
+            for (auto tf : ts->components) {
+                if (!tf->is<IR::Type_Bits>()) {
+                    ::error(ErrorType::ERR_UNSUPPORTED_ON_TARGET,
+                            "%1%: only integral values supported for logged values", mc);
+                    return primitive;
+                }
+            }
+        } else {
+            ::error(ErrorType::ERR_UNSUPPORTED_ON_TARGET,
+                    "%1%: Second argument must be a list expression: %2%", mc, arg1);
+            return primitive;
+        }
+
+        auto val = ctxt->conv->convert(arg1);
```

Comment:
can you call `convert` (https://github.com/p4lang/p4c/blob/master/backends/bmv2/common/expression.h#L105) with `convertBool` set to `true`? Then it will work fine with boolean values as well.

Reply:
Did that, but since this is a list I had to iterate over the elements first.